test/mind_the_gapp_web/controllers/arrival_time_controller_test.exs

```diff
@@ -0,0 +1,68 @@
+defmodule MindTheGappWeb.ArrivalTimeControllerTest do
+  use MindTheGappWeb.ConnCase, async: true
+  use Timex
+
+  setup_all :create_fake_arrival_time_data
+
+  def create_fake_arrival_time_data(_context) do
+    clear_database()
+    generate_fake_arrival_data()
+    :ok
+  end
+
+  def clear_database() do
+    MindTheGapp.Repo.delete_all(MindTheGapp.ArrivalTime)
+  end
+
+  def generate_fake_arrival_data() do
+    starting_date = DateTime.utc_now
+
+     1..50
+     |> Enum.reduce(starting_date, &persist_arrival_time/2)
+  end
+
+  def persist_arrival_time(_x, date_time) do
+    actual_time = DateTime.add(date_time, Enum.random(-10..10) * 60)
+
+     %MindTheGapp.ArrivalTime{estimated: date_time, actual: actual_time}
+     |> MindTheGapp.Repo.insert!()
+
+    date_time |> DateTime.add(3600 * -6)
+  end
+  describe "Arrival Time Controller" do
+    test "GET / returns a JSON object", %{conn: conn} do
+      conn = get(conn, "/")
+      assert json_response(conn, 200)
+    end
+
+    test "GET / returns a list of JSON objects whose attributes are 'day' and 'meanArrivalDifferences'", %{conn: conn} do
+      expected_attributes = ["day","meanArrivalDifferences"]
+      conn = get(conn, "/")
+      {:ok, retrieved_objects} = conn.resp_body
+                                 |> Jason.decode
+      retrieved_keys = retrieved_objects
+                       |> List.first
+                       |> Map.keys
+      assert retrieved_keys == expected_attributes
+    end
+
+    test "GET / returns a list of JSON objects, the 'meanArrivalDifferences' attributes are floats", %{conn: conn} do
```

Comment:
Do we care about this? I wonder if the query test is sufficient?

For these types of controller/integration tests I like to keep the tests focused on what the controller might do.
* Does the data match the *shape* I expect
* What happens when give it bad data/input (for POST request)
* How does it respond to edge cases (i.e. no data for the last 7 days).

Reply:
I thought it would test the shape of the data.